bitcoin/src/crypto/ecdsa.rs

```diff
@@ -98,6 +102,58 @@ impl FromStr for Signature {
     }
 }
 
+impl TryFrom<&str> for Signature {
+    type Error = Error;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        let bytes = Vec::from_hex(s)?;
+        let (sighash_byte, signature) = bytes.split_last().ok_or(Error::EmptySignature)?;
+        Ok(Signature {
+            signature: secp256k1::ecdsa::Signature::from_der(signature)?,
+            sighash_type: EcdsaSighashType::from_standard(*sighash_byte as u32)?,
+        })
+    }
+}
+
+impl TryFrom<Box<str>> for Signature {
+    type Error = Error;
+
+    fn try_from(s: Box<str>) -> Result<Self, Self::Error> {
+        let bytes = Vec::from_hex(&s)?;
+        let (sighash_byte, signature) = bytes.split_last().ok_or(Error::EmptySignature)?;
+        Ok(Signature {
+            signature: secp256k1::ecdsa::Signature::from_der(signature)?,
+            sighash_type: EcdsaSighashType::from_standard(*sighash_byte as u32)?,
+        })
+    }
+}
+
+impl TryFrom<Arc<str>> for Signature {
+    type Error = Error;
+
+    fn try_from(s: Arc<str>) -> Result<Self, Self::Error> {
+        let bytes = Vec::from_hex(&s)?;
+        let (sighash_byte, signature) = bytes.split_last().ok_or(Error::EmptySignature)?;
+        Ok(Signature {
+            signature: secp256k1::ecdsa::Signature::from_der(signature)?,
+            sighash_type: EcdsaSighashType::from_standard(*sighash_byte as u32)?,
+        })
+    }
+}
+
+impl TryFrom<Rc<str>> for Signature {
+    type Error = Error;
+
+    fn try_from(s: Rc<str>) -> Result<Self, Self::Error> {
+        let bytes = Vec::from_hex(&s)?;
+        let (sighash_byte, signature) = bytes.split_last().ok_or(Error::EmptySignature)?;
+        Ok(Signature {
+            signature: secp256k1::ecdsa::Signature::from_der(signature)?,
+            sighash_type: EcdsaSighashType::from_standard(*sighash_byte as u32)?,
+        })
+    }
+}
```

Comment:
Can you make the helper function and call it? Also maybe I'm wrong with the `InputString` part, I'm not sure what `Vec::from_hex` does.